### PR TITLE
Add new Insert Before/After menu items and shortcuts

### LINKED ISSUE #7297: Add "Insert Before" and "Insert After" buttons in the block More menu
Mockup:

mockup

The sibling inserter is receiving improvements. But in addition to this, it would be nice with a menu-discoverable way to insert blocks before and after the selected block, and even nicer if there were keyboard shortcuts to do so. 

See also discussion in https://github.com/WordPress/gutenberg/pull/7220#issuecomment-396886435 and https://github.com/WordPress/gutenberg/issues/7168.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Closes #7297

Adds the following keyboard shortcuts for those actions:
- <kbd>Option</kbd>+<kbd>Cmd</kbd>+<kbd>T</kbd> (Mac) / <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>T</kbd> - Insert Before
- <kbd>Option</kbd>+<kbd>Cmd</kbd>+<kbd>Y</kbd> (Mac) / <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>Y</kbd> - Insert After

## How has this been tested?
- Tested use of MenuItem and shortcut across browsers on Mac (Chrome, Safari, Firefox, Opera)
- Tested use of MenuItem and shortcut across browsers on Windows (Chrome, IE11, Edge, Firefox, Opera)
- Tested with a block that uses InnerBlocks with a template lock

## Screenshots 
**MenuItems**
screen shot 2018-08-13 at 4 33 36 pm

**Shortcut Help**
screen shot 2018-08-13 at 4 33 01 pm

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Review by jasmussen — APPROVED** [review]
Screenshot:

screen shot 2018-08-13 at 11 24 33

Love it. Works as intended.

Might want a code sanity check by @noisysocks, but otherwise, ship it!

**Comment by talldan** [comment]
Great - I still need to do a bit of work on the template locking side of things. I'll do that first and then get a code review

**Comment by talldan** [comment]
Ready for review now :)

File: packages/editor/src/components/block-settings-menu/index.js [hunk 1]
```diff
@@ -199,15 +235,32 @@ export default compose( [
...
+	withDispatch( ( dispatch, props ) => {
+		const {
+			clientIds,
+			rootClientId,
+			blocks,
+			firstSelectedIndex,
+			lastSelectedIndex,
+			isLocked,
+			canDuplicate,
+		} = props;
```
Comment (noisysocks) [inline packages/editor/src/components/block-settings-menu/index.js]: Nit: you could put newlines into these function arguments if you don't want `props` in the scope.

```js
withDispatch( ( dispatch, {
	clientIds,
	rootClientId,
	// etc.
} ) ) {
	// etc.
}
```

File: packages/editor/src/components/block-settings-menu/index.js [hunk 2]
```diff
@@ -229,7 +282,19 @@ export default compose( [
 				}
 			},
 			onRemove() {
-				removeBlocks( clientIds );
+				if ( ! isLocked ) {
+					removeBlocks( clientIds );
+				}
```
Comment (noisysocks) [inline packages/editor/src/components/block-settings-menu/index.js]: Nice, a sneaky bug fix!

**Review by noisysocks — APPROVED** [review]
Code looks fab, and everything works great in my testing!

I tested template locking using this custom post type:

```php
register_post_type( 'test_type', array(
	'label' => 'Test',
	'public' => true,
	'show_in_rest' => true,
	'template_lock' => 'insert',
	'template' => array(
		array( 'core/image' ),
		array( 'core/quote' ),
		array( 'core/image' ),
	),
) );
```

And this custom block:

```js
registerBlockType( 'test/block', {
	title: 'Test block',
	category: 'common',
	save: () => null,
	edit() {
		return (
			<wp.editor.InnerBlocks
				templateLock="insert"
				template={ [ [ 'core/image' ], [ 'core/image' ] ] }
			/>
		);
	},
} );
```

Great job 👍👍👌:shipit:

**Comment by mtias** [comment]
Nice one!

**Pull request merged**